checks.d/elastic.py

```diff
@@ -227,14 +244,75 @@ class ESCheck(AgentCheck):
+        "elasticsearch.thread_pool.listener.queue": ("gauge", "thread_pool.listener.queue"),
+        "elasticsearch.thread_pool.listener.rejected": ("gauge", "thread_pool.listener.rejected"),
+    }
+
+    ADDITIONAL_METRICS_POST_1_5_0 = {
+        "elasticsearch.indices.recovery.current_as_source": ("gauge", "indices.recovery.current_as_source"),
+        "elasticsearch.indices.recovery.current_as_target": ("gauge", "indices.recovery.current_as_target"),
+        "elasticsearch.indices.recovery.throttle_time_in_millis": ("gauge", "indices.recovery.throttle_time_in_millis", lambda v: float(v)/1000),
+    }
+
+    ADDITIONAL_METRICS_POST_1_6_0 = {
+        "elasticsearch.thread_pool.fetch_shard_started.active": ("gauge", "thread_pool.fetch_shard_started.active"),
+        "elasticsearch.thread_pool.fetch_shard_started.threads": ("gauge", "thread_pool.fetch_shard_started.threads"),
+        "elasticsearch.thread_pool.fetch_shard_started.queue": ("gauge", "thread_pool.fetch_shard_started.queue"),
+        "elasticsearch.thread_pool.fetch_shard_started.rejected": ("gauge", "thread_pool.fetch_shard_started.rejected"),
+        "elasticsearch.thread_pool.fetch_shard_store.active": ("gauge", "thread_pool.fetch_shard_store.active"),
+        "elasticsearch.thread_pool.fetch_shard_store.threads": ("gauge", "thread_pool.fetch_shard_store.threads"),
+        "elasticsearch.thread_pool.fetch_shard_store.queue": ("gauge", "thread_pool.fetch_shard_store.queue"),
+        "elasticsearch.thread_pool.fetch_shard_store.rejected": ("gauge", "thread_pool.fetch_shard_store.rejected"),
     }
 
     ADDITIONAL_METRICS_PRE_2_0 = {
         "elasticsearch.thread_pool.merge.active": ("gauge", "thread_pool.merge.active"),
         "elasticsearch.thread_pool.merge.threads": ("gauge", "thread_pool.merge.threads"),
         "elasticsearch.thread_pool.merge.queue": ("gauge", "thread_pool.merge.queue"),
+        "elasticsearch.thread_pool.merge.rejected": ("gauge", "thread_pool.merge.rejected"),
+    }
+
+    ADDITIONAL_METRICS_POST_2_0 = {
+        # Some of these may very well exist in previous ES versions, but not worth the time/effort
+        # to find where they were introduced
+        "elasticsearch.indices.query_cache.cache_size": ("gauge", "indices.query_cache.cache_size"),
+        "elasticsearch.indices.query_cache.cache_count": ("gauge", "indices.query_cache.cache_count"),
+        "elasticsearch.indices.query_cache.total_count": ("gauge", "indices.query_cache.total_count"),
+        "elasticsearch.indices.segments.doc_values_memory_in_bytes": ("gauge", "indices.segments.doc_values_memory_in_bytes"),
+        "elasticsearch.indices.segments.norms_memory_in_bytes": ("gauge", "indices.segments.norms_memory_in_bytes"),
+        "elasticsearch.indices.segments.stored_fields_memory_in_bytes": ("gauge", "indices.segments.stored_fields_memory_in_bytes"),
+        "elasticsearch.indices.segments.term_memory_in_bytes": ("gauge", "indices.segments.term_memory_in_bytes"),
```

Comment:
This stat `term_memory_in_bytes` is unavailable - might be a typo that was forgotten here. This is what's causing the two 2.0+ tests to fail. Removal of the metric works fine for me locally.